File: 1752. Check if Array Is Sorted and Rotated/Solution.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        if len(set(nums)) == 1: return True
        i, n = 1, len(nums)
        while i < n and nums[i] >= nums[i - 1]:
            i += 1
        if i == n: return True
        for j in range(i + 1, n):
            if nums[j] < nums[j - 1]: return False
        return nums[0] >= nums[-1]

File: 1752. Check if Array Is Sorted and Rotated/Solution.java
class Solution {
    public boolean check(int[] nums) {
        int i = 1, n = nums.length;
        while (i < n && nums[i] >= nums[i - 1]) i++;
        if (i == n) return true;
        for (int j = i + 1; j < n; j++) {
            if (nums[j] < nums[j - 1]) return false;
        }
        return nums[0] >= nums[n - 1];
    }
}